agent/whois_ip_agent.py

```diff
@@ -54,19 +55,29 @@ def process(self, message: m.Message) -> None:
 
         else:
             host = message.data.get('host')
-            if not self.set_add('agent_whois_ip_asset', host):
-                logger.info('target %s was processed before, exiting', host)
-                return
-            try:
-                logger.info('processing IP %s', host)
-                record = ipwhois.IPWhois(host).lookup_rdap()
-                mask = message.data.get('mask')
-                version = message.data.get('version')
-                whois_message = ipwhois_data_handler.prepare_whois_message_data(host, mask, version, record)
-                self._emit_whois_message(version, whois_message)
-            except ipwhois.exceptions.IPDefinedError as e:
-                # casewhere of loopback address
-                logger.error('%s', e)
+            mask = message.data.get('mask')
+
+            if mask is not None:
+                addresses = ipaddress.ip_network(f'{host}/{mask}')
```

Comment:
If there is no mask, it is better to handle it as a /32 for ipv4 and /164 for ipv6.